### PR TITLE
Update version to '0.3.1-preview1'.

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This PR updates the version, and pulls in 2 commits that are on main, which would be nice to have on 0.3.x as well:
fixes to the language extension settings.

This will cause some merge conflicts with my other PR, which I'll deal with.

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -70,5 +70,8 @@
     "rust",
     "softwareTerms",
     "en_US"
+  ],
+  "enableFiletypes": [
+    "slice"
   ]
```
Comment (pepone) [inline .vscode/cspell.json:76]: This seems unrelated to the PR, is it required?

**Review by pepone — APPROVED** [review]

File: .vscode/cspell.json (hunk 1 above)
Comment (InsertCreativityHere) [reply .vscode/cspell.json:76]: You got here before I edited the comment!

> This PR updates the version, and pulls in 2 commits that are on main, which would be nice to have on 0.3.x as well:
fixes to the language extension settings.

Comment (InsertCreativityHere) [reply .vscode/cspell.json:76]: I was going to rebase this PR to pull them along, but keep the commits seperate.

**Review by bernardnormier — APPROVED** [review]
